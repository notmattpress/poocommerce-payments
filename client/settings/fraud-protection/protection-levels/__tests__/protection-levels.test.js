/**
 * External dependencies
 */
import React from 'react';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import ProtectionLevels from '..';

let mockFraudProtectionRule = [];

jest.mock( 'wcpay/data', () => ( {
	...jest.requireActual( 'wcpay/data' ),
	useAdvancedFraudProtectionSettings: jest.fn( () => [
		mockFraudProtectionRule,
		jest.fn(),
	] ),
} ) );

describe( 'ProtectionLevels', () => {
	beforeEach( () => {
		global.wcpaySettings = {
			isMultiCurrencyEnabled: '1',
		};
	} );

	it( 'renders', () => {
		mockFraudProtectionRule = [];
		const { container: protectionLevels } = render( <ProtectionLevels /> );

		expect( protectionLevels ).toMatchSnapshot();
	} );

	it( 'renders with the advanced fraud protection settings selected', () => {
		mockFraudProtectionRule = [
			{
				key: 'international_ip_address',
				outcome: 'review',
				check: {
					key: 'ip_country',
					operator: 'in',
					value: '',
				},
			},
		];
		const { container: protectionLevels } = render( <ProtectionLevels /> );

		expect( protectionLevels ).toMatchSnapshot();
	} );

	it( 'renders an error message when settings can not be fetched from the server', () => {
		mockFraudProtectionRule = 'error';
		const { container: protectionLevels } = render( <ProtectionLevels /> );

		expect( protectionLevels ).toMatchSnapshot();
		expect( protectionLevels ).toHaveTextContent(
			/There was an error retrieving your fraud protection settings/i
		);
		expect(
			protectionLevels.getElementsByTagName( 'fieldset' )[ 0 ]
		).toBeDisabled();
	} );
} );
