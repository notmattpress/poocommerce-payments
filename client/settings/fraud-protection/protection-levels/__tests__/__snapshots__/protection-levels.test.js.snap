// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectionLevels renders 1`] = `
<div>
  <fieldset
    id="fraud-protection-card-options"
  >
    <ul>
      <li>
        <div
          class="fraud-protection-radio-wrapper"
        >
          <input
            checked=""
            class="fraud-protection-radio-wrapper__item"
            id="fraud-protection__basic-level"
            name="fraud-protection-level-select"
            type="radio"
            value="basic"
          />
          <label
            class="fraud-protection-radio-wrapper__item"
            for="fraud-protection__basic-level"
          >
            Basic
          </label>
          <svg
            class="gridicon gridicons-help-outline fraud-protection__help-icon needs-offset"
            height="18"
            title="Basic level help icon"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm4 8a4 4 0 00-8 0h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2a1 1 0 00-1 1v2h2v-1.141A3.991 3.991 0 0016 10zm-3 6h-2v2h2v-2z"
              />
            </g>
          </svg>
        </div>
        <p
          class="fraud-protection__text--help-text"
        >
          Provides the base level of platform protection.
        </p>
      </li>
      <li
        class="fraud-protection__advanced-level-container"
      >
        <label
          for="fraud-protection-level-select_advanced-level"
        >
          <div
            class="fraud-protection-radio-wrapper"
          >
            <input
              id="fraud-protection-level-select_advanced-level"
              name="fraud-protection-level-select"
              type="radio"
              value="advanced"
            />
            <p
              class="fraud-protection-radio-wrapper__item"
            >
              Advanced
            </p>
          </div>
          <p
            class="fraud-protection__text--help-text"
          >
            Allows you to fine-tune the level of filtering according to your business needs.
          </p>
        </label>
        <button
          class="components-button is-secondary"
          disabled=""
          type="button"
        >
          Configure
        </button>
      </li>
    </ul>
  </fieldset>
</div>
`;

exports[`ProtectionLevels renders an error message when settings can not be fetched from the server 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: calc(4px * 2);
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  width: 100%;
}

.emotion-2>* {
  min-width: 0;
}

.emotion-4 {
  display: block;
  max-height: 100%;
  max-width: 100%;
  min-height: 0;
  min-width: 0;
}

<div>
  <div
    class="wcpay-inline-notice wcpay-inline-error-notice components-notice is-error"
  >
    <div
      class="components-visually-hidden emotion-0 emotion-1"
      data-wp-c16t="true"
      data-wp-component="VisuallyHidden"
      style="border: 0px; clip: rect(1px, 1px, 1px, 1px); clip-path: inset( 50% ); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; word-wrap: normal;"
    >
      Error notice
    </div>
    <div
      class="components-notice__content"
    >
      <div
        class="components-flex emotion-2 emotion-1"
        data-wp-c16t="true"
        data-wp-component="Flex"
      >
        <div
          class="components-flex-item wcpay-inline-notice__icon wcpay-inline-error-notice__icon emotion-4 emotion-1"
          data-wp-c16t="true"
          data-wp-component="FlexItem"
        >
          <svg
            class="gridicon gridicons-notice-outline"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
              />
            </g>
          </svg>
        </div>
        <div
          class="components-flex-item wcpay-inline-notice__content wcpay-inline-error-notice__content emotion-4 emotion-1"
          data-wp-c16t="true"
          data-wp-component="FlexItem"
        >
          There was an error retrieving your fraud protection settings. Please refresh the page to try again.
        </div>
      </div>
      <div
        class="components-notice__actions"
      />
    </div>
  </div>
  <fieldset
    disabled=""
    id="fraud-protection-card-options"
  >
    <ul>
      <li>
        <div
          class="fraud-protection-radio-wrapper"
        >
          <input
            checked=""
            class="fraud-protection-radio-wrapper__item"
            id="fraud-protection__basic-level"
            name="fraud-protection-level-select"
            type="radio"
            value="basic"
          />
          <label
            class="fraud-protection-radio-wrapper__item"
            for="fraud-protection__basic-level"
          >
            Basic
          </label>
          <svg
            class="gridicon gridicons-help-outline fraud-protection__help-icon needs-offset"
            height="18"
            title="Basic level help icon"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm4 8a4 4 0 00-8 0h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2a1 1 0 00-1 1v2h2v-1.141A3.991 3.991 0 0016 10zm-3 6h-2v2h2v-2z"
              />
            </g>
          </svg>
        </div>
        <p
          class="fraud-protection__text--help-text"
        >
          Provides the base level of platform protection.
        </p>
      </li>
      <li
        class="fraud-protection__advanced-level-container"
      >
        <label
          for="fraud-protection-level-select_advanced-level"
        >
          <div
            class="fraud-protection-radio-wrapper"
          >
            <input
              id="fraud-protection-level-select_advanced-level"
              name="fraud-protection-level-select"
              type="radio"
              value="advanced"
            />
            <p
              class="fraud-protection-radio-wrapper__item"
            >
              Advanced
            </p>
          </div>
          <p
            class="fraud-protection__text--help-text"
          >
            Allows you to fine-tune the level of filtering according to your business needs.
          </p>
        </label>
        <button
          class="components-button is-secondary"
          disabled=""
          type="button"
        >
          Configure
        </button>
      </li>
    </ul>
  </fieldset>
</div>
`;

exports[`ProtectionLevels renders with the advanced fraud protection settings selected 1`] = `
<div>
  <fieldset
    id="fraud-protection-card-options"
  >
    <ul>
      <li>
        <div
          class="fraud-protection-radio-wrapper"
        >
          <input
            checked=""
            class="fraud-protection-radio-wrapper__item"
            id="fraud-protection__basic-level"
            name="fraud-protection-level-select"
            type="radio"
            value="basic"
          />
          <label
            class="fraud-protection-radio-wrapper__item"
            for="fraud-protection__basic-level"
          >
            Basic
          </label>
          <svg
            class="gridicon gridicons-help-outline fraud-protection__help-icon needs-offset"
            height="18"
            title="Basic level help icon"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm4 8a4 4 0 00-8 0h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2a1 1 0 00-1 1v2h2v-1.141A3.991 3.991 0 0016 10zm-3 6h-2v2h2v-2z"
              />
            </g>
          </svg>
        </div>
        <p
          class="fraud-protection__text--help-text"
        >
          Provides the base level of platform protection.
        </p>
      </li>
      <li
        class="fraud-protection__advanced-level-container"
      >
        <label
          for="fraud-protection-level-select_advanced-level"
        >
          <div
            class="fraud-protection-radio-wrapper"
          >
            <input
              id="fraud-protection-level-select_advanced-level"
              name="fraud-protection-level-select"
              type="radio"
              value="advanced"
            />
            <p
              class="fraud-protection-radio-wrapper__item"
            >
              Advanced
            </p>
          </div>
          <p
            class="fraud-protection__text--help-text"
          >
            Allows you to fine-tune the level of filtering according to your business needs.
          </p>
        </label>
        <button
          class="components-button is-secondary"
          disabled=""
          type="button"
        >
          Edit
        </button>
      </li>
    </ul>
  </fieldset>
</div>
`;
